integration_tests/src/main/python/hive_delimited_text_test.py

```diff
@@ -270,24 +272,38 @@ def test_hive_text_round_trip_partitioned(spark_tmp_path, data_gen, spark_tmp_ta
     ArrayGen(string_gen),
     StructGen([('int_field', int_gen), ('string_field', string_gen)]),
     MapGen(StringGen(nullable=False), string_gen),
-    binary_gen
+    binary_gen,
+    StructGen([('b', byte_gen), ('i', int_gen), ('arr_of_i', ArrayGen(int_gen))]),
+    ArrayGen(StructGen([('b', byte_gen), ('i', int_gen), ('arr_of_i', ArrayGen(int_gen))]))
 ]
 
 
 @allow_non_gpu("org.apache.spark.sql.hive.execution.HiveTableScanExec")
 @pytest.mark.parametrize('data_gen', hive_text_unsupported_gens, ids=idfn)
 def test_hive_text_fallback_for_unsupported_types(spark_tmp_path, data_gen, spark_tmp_table_factory):
-    gen = StructGen([('my_field', data_gen)], nullable=False)
+    gen = StructGen([('my_supported_int_field', int_gen),
+                     ('my_unsupported_field', data_gen), ], nullable=False)
```

Comment:
nit, if `data_gen` is all about unsupported types, think we should call it `unsupported_data_gen` or something like that. Perhaps `int_gen` could then also be renamed to `supported_data_gen`, and it can default to `int_gen`.

Reply:
Agreed. That does read better.